crates/pallets/starknet/src/lib.rs

```diff
@@ -1106,6 +1106,28 @@ impl<T: Config> Pallet<T> {
         Ok(results)
     }
 
+    pub fn estimate_message_fee(message: HandleL1MessageTransaction) -> Result<(u64, u64), DispatchError> {
```

Comment:
Add some warning, saying that this should only be called from the runtimeAPI, never from inside an extrinsic execution flow coz i WILL execute the message